ocpp-common/src/main/java/eu/chargetime/ocpp/Client.java

```diff
@@ -44,22 +44,20 @@ of this software and associated documentation files (the "Software"), to deal
 public class Client {
   private static final Logger logger = LoggerFactory.getLogger(Client.class);
 
-  private ISession session;
-  private final IFeatureRepository featureRepository;
+  private final ISession session;
   private final IPromiseRepository promiseRepository;
 
   /**
    * Handle required injections.
    *
    * @param session Inject session object
+   * @param promiseRepository Inject promise repository
    * @see Session
    */
   public Client(
       ISession session,
-      IFeatureRepository featureRepository,
```

Comment:
Does this really break anything? Applications using JSONClient and JSONServer continue to work without modifications, see FakeCentralSystem and FakeChargePoint (the latter has only been modified to make part of its implementation accessible to the new multi-protocol tests). Has any application ever fiddled with directly constructing Client or Session objects? Are these even parts of the public API?

I kept the version at 1.x to indicate full backwards compatibility. But if there are indeed applications which no longer work with these changes, I'd prefer to bump the main branch to version 2.0.

Reply:
You're probably right, it's been too long since I worked with the library. I'll just have to double check. 

If an application has made changes to the dependency injection or lower level APIs, they probably won't update to the most recent version automatically. In fact they should probably fork the library and maintain their own branch.